server/adaptors/integrations/integrations_builder.ts

```diff
@@ -28,32 +29,44 @@ export class IntegrationInstanceBuilder {
     this.client = client;
   }
 
-  build(integration: IntegrationReader, options: BuilderOptions): Promise<IntegrationInstance> {
-    const instance = deepCheck(integration)
-      .then((result) => {
-        if (!result.ok) {
-          return Promise.reject(result.error);
+  async build(
+    integration: IntegrationReader,
+    options: BuilderOptions
+  ): Promise<IntegrationInstance> {
+    const instance = await deepCheck(integration);
+    if (!instance.ok) {
+      return Promise.reject(instance.error);
+    }
+    const assets = await integration.getAssets();
+    if (!assets.ok) {
+      return Promise.reject(assets.error);
+    }
+    const remapped = this.remapIDs(this.getSavedObjectBundles(assets.value));
+    const withDataSource = this.remapDataSource(remapped, options.dataSource);
+    const refs = await this.postAssets(withDataSource);
+    const builtInstance = await this.buildInstance(integration, refs, options);
+    return builtInstance;
+  }
+
+  getSavedObjectBundles(
```

Comment:
This is the core of the backend changes -- There are several small type changes that lead to this method filtering assets based on what's actually included in the current bundles.